src/main/scala/com/etsy/sbt/Checkstyle.scala

```diff
@@ -8,39 +8,77 @@ import sbt.Def.Initialize
 import sbt.Keys._
 import sbt._
 
+import scala.io.Source
+
 /**
   * An SBT plugin to run checkstyle over Java code
   *  
   * @author Andrew Johnson <ajohnson@etsy.com>
   */
 object Checkstyle extends Plugin {
-  import com.etsy.sbt.Checkstyle.CheckstyleTasks._
+
+  sealed abstract class CheckstyleConfig(val location: String) {
+    def read(resources: Seq[File]): String
+  }
+
+  object CheckstyleConfig {
+    case class URL(url: String) extends CheckstyleConfig(url) {
+      override def read(resources: Seq[sbt.File]): String = Source.fromURL(url).mkString
+    }
+
+    case class File(path: String) extends CheckstyleConfig(path) {
+      override def read(resources: Seq[sbt.File]): String = Source.fromFile(path).mkString
+    }
+
+    case class Classpath(name: String) extends CheckstyleConfig(name) {
+      override def read(resources: Seq[sbt.File]): String = {
+        val classpath = resources.map((f) => f.toURI.toURL)
```

Comment:
I'm not sure that this will support loading files out of JARs on the classpath.  I was experimenting with `resources in Compile` and it looks like it only includes resource files that are local to the build.  If you do `(fullClasspath in Compile).value.map(_.data)` you'll get a `Seq[File]` of everything that would be on the classpath that you could then use to construct the `URLClassLoader`.